From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lumine1909 <133463833+Lumine1909@users.noreply.github.com>
Date: Sat, 17 May 2025 17:36:22 +0800
Subject: [PATCH 04/10] Replay Mod API


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 6997c8cf4..37b2231a7 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -3016,4 +3016,10 @@ public final class Bukkit {
         return server.getBotManager();
     }
     // Leaves end - Bot API
+
+    // Leaves start - Photographer API
+    public static @NotNull org.leavesmc.leaves.entity.photographer.PhotographerManager getPhotographerManager() {
+        return server.getPhotographerManager();
+    }
+    // Leaves end - Photographer API
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 42d097c41..f343cba49 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2783,4 +2783,6 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     @NotNull org.leavesmc.leaves.entity.bot.BotManager getBotManager();
     // Leaves end - Bot API
+
+    @NotNull org.leavesmc.leaves.entity.photographer.PhotographerManager getPhotographerManager(); // Leaves - Photographer API
 }
