From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesX Build <build@leavesx.local>
Date: Thu, 13 Aug 2026 13:15:09 +0800
Subject: [PATCH 10/10] Declare Leaves API accessors non-null


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 318a3e7c1..cf73e668c 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -3007,7 +3007,7 @@ public final class Bukkit {
     }
 
     // Leaves start - Feature API
-    public static org.leavesmc.leaves.plugin.FeatureManager getFeatureManager() {
+    public static @NotNull org.leavesmc.leaves.plugin.FeatureManager getFeatureManager() {
         return server.getFeatureManager();
     }
     // Leaves end - Feature API
@@ -3030,13 +3030,13 @@ public final class Bukkit {
     // Leaves end - Photographer API
 
     // Leaves start - Bytebuf API
-    public static org.leavesmc.leaves.bytebuf.BytebufAllocator getBytebufManager() {
+    public static @NotNull org.leavesmc.leaves.bytebuf.BytebufAllocator getBytebufManager() {
         return server.getBytebufAllocator();
     }
     // Leaves end - Bytebuf API
 
     // Leaves start - Config API
-    public static org.leavesmc.leaves.config.LeavesConfigProvider getLeavesConfig() {
+    public static @NotNull org.leavesmc.leaves.config.LeavesConfigProvider getLeavesConfig() {
         return server.getLeavesConfig();
     }
     // Leaves end - Config API
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 4e548f3c1..27658c805 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2776,7 +2776,7 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     // Paper end - API to check if the server is sleeping
 
     // Leaves start - Feature API
-    org.leavesmc.leaves.plugin.FeatureManager getFeatureManager();
+    @NotNull org.leavesmc.leaves.plugin.FeatureManager getFeatureManager();
     // Leaves End - Feature API
 
     // Leaves start - Bot API
@@ -2790,7 +2790,7 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
 
     @NotNull org.leavesmc.leaves.entity.photographer.PhotographerManager getPhotographerManager(); // Leaves - Photographer API
 
-    org.leavesmc.leaves.bytebuf.BytebufAllocator getBytebufAllocator(); // Leaves - Bytebuf API
+    @NotNull org.leavesmc.leaves.bytebuf.BytebufAllocator getBytebufAllocator(); // Leaves - Bytebuf API
 
-    org.leavesmc.leaves.config.LeavesConfigProvider getLeavesConfig(); // Leaves - Config API
+    @NotNull org.leavesmc.leaves.config.LeavesConfigProvider getLeavesConfig(); // Leaves - Config API
 }
