From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lumine1909 <133463833+Lumine1909@users.noreply.github.com>
Date: Tue, 22 Apr 2025 23:32:41 +0800
Subject: [PATCH] Leaves Utils


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index d958c471..31c69d61 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -388,6 +388,7 @@ public abstract class Entity
     public long activatedTick = Integer.MIN_VALUE;
     public boolean isTemporarilyActive;
     public long activatedImmunityTick = Integer.MIN_VALUE;
+    private net.minecraft.nbt.CompoundTag leavesData = new net.minecraft.nbt.CompoundTag(); // Leaves - Leaves ex data
 
     public void inactiveTick() {
     }
@@ -2774,6 +2775,7 @@ public abstract class Entity
                 output.putBoolean("Paper.FreezeLock", true);
             }
             // Paper end
+            output.store("Leaves.Data", net.minecraft.nbt.CompoundTag.CODEC, leavesData); // Leaves - leaves ex data
         } catch (Throwable t) {
             CrashReport report = CrashReport.forThrowable(t, "Saving entity NBT");
             CrashReportCategory category = report.addCategory("Entity being saved");
@@ -2893,6 +2895,7 @@ public abstract class Entity
             }
             freezeLocked = input.getBooleanOr("Paper.FreezeLock", false);
             // Paper end
+            leavesData = input.read("Leaves.Data", net.minecraft.nbt.CompoundTag.CODEC).orElse(new net.minecraft.nbt.CompoundTag());
         } catch (Throwable t) {
             CrashReport report = CrashReport.forThrowable(t, "Loading entity NBT");
             CrashReportCategory category = report.addCategory("Entity being loaded");
@@ -4665,7 +4668,7 @@ public abstract class Entity
         return this.is(EntityTypeTags.DEFLECTS_PROJECTILES) ? ProjectileDeflection.REVERSE : ProjectileDeflection.NONE;
     }
 
-    public @Nullable LivingEntity getControllingPassenger() {
+    public @org.jetbrains.annotations.NotNull LivingEntity getControllingPassenger() { // Leaves - notnull
         return null;
     }
 
@@ -5469,5 +5472,11 @@ public abstract class Entity
     public boolean isTicking() {
         return ((ServerLevel) this.level()).isPositionEntityTicking(this.blockPosition());
     }
-    // Paper end
+    // Paper end - Expose entity id counter
+
+    // Leaves start - leaves ex data
+    public net.minecraft.nbt.CompoundTag getLeavesData() {
+        return leavesData;
+    }
+    // Leaves end - leaves ex data
 }
diff --git a/net/minecraft/world/level/LevelAccessor.java b/net/minecraft/world/level/LevelAccessor.java
index 700b9e7f..b7c52129 100644
--- a/net/minecraft/world/level/LevelAccessor.java
+++ b/net/minecraft/world/level/LevelAccessor.java
@@ -42,7 +42,7 @@ public interface LevelAccessor extends CommonLevelAccessor, ScheduledTickAccess
         return this.getLevelData().getGameTime();
     }
 
-    @Nullable MinecraftServer getServer();
+    @org.jetbrains.annotations.NotNull MinecraftServer getServer(); // Leaves - notnull
 
     default Difficulty getDifficulty() {
         return this.getLevelData().getDifficulty();
diff --git a/net/minecraft/world/level/block/Block.java b/net/minecraft/world/level/block/Block.java
index 29f5182c..debf6137 100644
--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -696,6 +696,13 @@ public class Block extends BlockBehaviour implements ItemLike {
     }
     // CraftBukkit end
 
+    // Leaves start - reset push reaction
+    @org.jetbrains.annotations.Nullable
+    public net.minecraft.world.level.material.PushReaction getResetPushReaction() {
+        return null;
+    }
+    // Leaves end - reset push reaction
+
     private record ShapePairKey(VoxelShape first, VoxelShape second) {
         @Override
         public boolean equals(final Object o) {
diff --git a/net/minecraft/world/level/block/state/BlockBehaviour.java b/net/minecraft/world/level/block/state/BlockBehaviour.java
index c78d7708..af470327 100644
--- a/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -792,7 +792,7 @@ public abstract class BlockBehaviour implements FeatureElement {
         }
 
         public PushReaction getPistonPushReaction() {
-            return !this.isDestroyable() ? PushReaction.BLOCK : this.pushReaction; // Paper - Protect Bedrock and End Portal/Frames from being destroyed
+            return !this.isDestroyable() ? PushReaction.BLOCK : this.getBlock().getResetPushReaction() == null ? this.pushReaction : this.getBlock().getResetPushReaction(); // Paper - Protect Bedrock and End Portal/Frames from being destroyed // Leaves - reset push reaction
         }
 
         public boolean isSolidRender() {
