From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: violetc <58360096+s-yh-china@users.noreply.github.com>
Date: Thu, 2 Jan 2025 22:01:17 +0800
Subject: [PATCH] Fix EntityPortalExitEvent logic


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index b5803102..aa01c208 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -4170,14 +4170,15 @@ public abstract class Entity
 
                 // Only change the target if actually needed, since we reset relative flags
                 if (event.isCancelled() || !to.equals(event.getTo()) || !after.equals(event.getAfter())) {
-                    if (event.isCancelled() || event.getTo() == null) {
-                        org.bukkit.World toWorld = to.getWorld();
-                        to = event.getFrom().clone();
-                        to.setWorld(toWorld); // cancelling doesn't cancel the teleport just the position/velocity (old quirk)
-                        velocity = org.bukkit.craftbukkit.util.CraftVector.toVec3(event.getBefore());
-                    } else {
-                        to = event.getTo().clone();
+                    // Leaves - cancelling a portal exit keeps the entity in its original world.
+                    if (!event.isCancelled()) {
+                        if (event.getTo() != null) {
+                            to = event.getTo().clone();
+                        }
                         velocity = org.bukkit.craftbukkit.util.CraftVector.toVec3(event.getAfter());
+                    } else {
+                        to = event.getFrom().clone();
+                        velocity = Vec3.ZERO;
                     }
                     transition = transition.withChangedLocation(to).withRelatives(Set.of()).withVelocity(velocity);
                 }
