From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MC_XiaoHei <xor7xiaohei@gmail.com>
Date: Sat, 2 Aug 2025 14:38:52 +0800
Subject: [PATCH] Bring back LivingEntity effect CME


diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index c996f752..1ee9e5ba 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1159,7 +1159,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
 
     public boolean addEffect(final MobEffectInstance newEffect, final @Nullable Entity source, final EntityPotionEffectEvent.Cause cause, final boolean fireEvent) {
         // Paper end - Don't fire sync event during generation
-        if (this.tickingEffects) {
+        if (this.tickingEffects && cause == org.bukkit.event.entity.EntityPotionEffectEvent.Cause.PLUGIN) { // Leaves - bring back LivingEntity effect CME
             this.deferredEffectUpdates.offer(() -> this.addEffect(newEffect, source, cause, fireEvent));
             return true;
         }
