From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lumine1909 <133463833+Lumine1909@users.noreply.github.com>
Date: Sat, 23 May 2026 14:32:26 +0800
Subject: [PATCH] Fix piercing weapon can attack in spectator mode


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index f81450f5..a2539312 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1956,7 +1956,7 @@ public class ServerGamePacketListenerImpl
                         }
 
                         PiercingWeapon piercingWeapon = itemInHand.get(DataComponents.PIERCING_WEAPON);
-                        if (piercingWeapon != null) {
+                        if (piercingWeapon != null && this.player.gameMode() != net.minecraft.world.level.GameType.SPECTATOR) { // Leaves - fix piercing weapon can attack in spectator mode
                             piercingWeapon.attack(this.player, EquipmentSlot.MAINHAND);
                         }
 
