From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesX Build <build@leavesx.local>
Date: Wed, 12 Aug 2026 23:34:36 +0800
Subject: [PATCH] Allow detached inventory capacity checks


diff --git a/net/minecraft/world/entity/player/Inventory.java b/net/minecraft/world/entity/player/Inventory.java
index 47e1a18c..686eefc6 100644
--- a/net/minecraft/world/entity/player/Inventory.java
+++ b/net/minecraft/world/entity/player/Inventory.java
@@ -179,7 +179,7 @@ public class Inventory implements Container, Nameable {
         int remains = itemStack.getCount();
 
         // Leaves start - vanilla creative pickup behavior
-        if (player.hasInfiniteMaterials()) {
+        if (this.player != null && this.player.hasInfiniteMaterials()) {
             return remains;
         }
         // Leaves end - vanilla creative pickup behavior
