From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesX Build <build@leavesx.local>
Date: Fri, 21 Aug 2026 20:18:23 +0800
Subject: [PATCH] Use activation snapshots for large entity batches in beta147


diff --git a/io/papermc/paper/entity/activation/ActivationRange.java b/io/papermc/paper/entity/activation/ActivationRange.java
index c8768007..a09b716b 100644
--- a/io/papermc/paper/entity/activation/ActivationRange.java
+++ b/io/papermc/paper/entity/activation/ActivationRange.java
@@ -166,6 +166,9 @@ public final class ActivationRange {
             ActivationType.VILLAGER.boundingBox = playerBoundingBox.inflate(villagerActivationRange, worldHeight, villagerActivationRange);
 
             final List<Entity> entities = world.getEntities((Entity) null, ActivationRange.maxBB, e -> true);
+            if (ActivationRangeSnapshot.activateEntities(entities, tickMarkers, currentTick)) {
+                continue;
+            }
             for (int entityIndex = 0, entityCount = entities.size(); entityIndex < entityCount; entityIndex++) {
                 final Entity entity = entities.get(entityIndex);
                 if (!tickMarkers && entity instanceof Marker) {
