From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesX Build <leavesx-build@local>
Date: Fri, 28 Aug 2026 10:00:21 +0800
Subject: [PATCH] Brand server as LeavesX


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
index f38f8e5851ec65a8f3f98018d55de74ea6ff165c..89119b023db4dc4d84574c877bacdd8788c10553 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -30,7 +30,7 @@ public record ServerBuildInfoImpl(
     private static final String ATTRIBUTE_GIT_BRANCH = "Git-Branch";
     private static final String ATTRIBUTE_GIT_COMMIT = "Git-Commit";
 
-    private static final String BRAND_PAPER_NAME = "Leaves"; // Leaves - Paper -> Leaves
+    private static final String BRAND_PAPER_NAME = "LeavesX"; // LeavesX - Paper -> Leaves
 
     private static final String BUILD_DEV = "DEV";
 
