From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesX Build <build@leavesx.local>
Date: Wed, 12 Aug 2026 18:44:23 +0800
Subject: [PATCH] Keep bundled MPL license bytes stable


diff --git a/.gitattributes b/.gitattributes
new file mode 100644
index 0000000000000000000000000000000000000000..85edd8b2babfa0f4c18467bc5ea2073f7a5ce381
--- /dev/null
+++ b/.gitattributes
@@ -0,0 +1 @@
+/src/main/resources/META-INF/licenses/leaves-documentation-MPL-2.0.txt -text
